### PR TITLE
support for GPU monitoring installation

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <talonzhu@yunify.com>

This pr aims to support GPU monitoring installation. Fix https://github.com/kubesphere/kubesphere/issues/4082

The notable fix can be described as follows:
- make definition in cluster-configuration yaml:
```
monitoring:
    gpu:
      nvidia_dcgm_exporter:
        enabled: true
```
-  Integrate a GPU monitoring task in the monitoring section, the steps can be found at `gpu-monitoring.yaml`.
   - Getting GPU monitoring installation files.
   - Creating GPU monitoring manifests.
   - Labeling GPU nodes.
   - Installing NVIDIA DCGM exporter( From [NVIDIA/gpu-monitoring-tools](https://github.com/NVIDIA/gpu-monitoring-tools)).
   - Installing custom GPU dashboards( From [nvidia-gpu-dcgm-exporter-clusterdashboard.yaml](https://github.com/kubesphere/monitoring-dashboard/blob/master/contrib/gallery/clusterdashboards/nvidia-gpu-dcgm-exporter-clusterdashboard.yaml)).
-  For the cluster role updates:
```
- apiGroups:
  - monitoring.kubesphere.io
  resources:
  - '*'
  verbs:
  - '*'
```

/cc @benjaminhuo @pixiake

### CODE REVIEW

File: deploy/cluster-configuration.yaml [hunk 1]
```diff
@@ -95,6 +95,9 @@ spec:
     prometheusMemoryRequest: 400Mi   # Prometheus request memory.
     prometheusVolumeSize: 20Gi       # Prometheus PVC size.
     # alertmanagerReplicas: 1          # AlertManager Replicas.
+    # gpu:                           # GPU monitoring-related plugins installation.
```
Comment (benjaminhuo) [inline deploy/cluster-configuration.yaml:98]: can we uncomment the entire GPU section and set nvidia_dcgm_exporter to false?

Comment (zhu733756) [reply deploy/cluster-configuration.yaml]: fixed

**Review by benjaminhuo — APPROVED** [review]
/lgtm

**Pull request merged**